src/Microsoft.DotNet.Build.Tasks/VersionTools/BaseDependenciesTask.cs

```diff
@@ -0,0 +1,163 @@
+                            i => i.Info,
+                            i => i.Skip.Equals("true", StringComparison.OrdinalIgnoreCase))
+                };
+            }
+
+            foreach (ITaskItem step in XmlUpdateStep ?? Enumerable.Empty<ITaskItem>())
+            {
+                string buildInfoName = step.GetMetadata("BuildInfoName");
+                string packageId = step.GetMetadata("PackageId");
+
+                FileRegexUpdater updater;
+
+                if (!string.IsNullOrEmpty(buildInfoName))
+                {
+                    updater = new FileRegexReleaseUpdater
+                    {
+                        BuildInfoName = buildInfoName
+                    };
+                }
+                else
+                {
+                    updater = new FileRegexPackageUpdater
+                    {
+                        PackageId = packageId
+                    };
+                }
+                updater.Path = step.GetMetadata("Path");
+                updater.Regex = CreateXmlUpdateRegex(step.GetMetadata("ElementName"), "version");
+                updater.VersionGroupName = "version";
+
+                yield return updater;
+            }
+        }
+
+        protected IEnumerable<BuildInfo> CreateBuildInfos()
+        {
+            return DependencyBuildInfo
+                ?.Select(item => CreateBuildInfo(item, BuildInfoCacheDir))
+                ?? Enumerable.Empty<BuildInfo>();
+        }
+
+        private static BuildInfo CreateBuildInfo(ITaskItem item, string cacheDir)
+        {
+            string rawUrl = item.GetMetadata("RawUrl");
+
+            if (!string.IsNullOrEmpty(rawUrl))
+            {
+                return BuildInfo.Get(item.ItemSpec, rawUrl);
+            }
+
+            string rawVersionsBaseUrl = item.GetMetadata("RawVersionsBaseUrl");
+            string buildInfoPath = item.GetMetadata("BuildInfoPath");
+            string currentRef = item.GetMetadata("CurrentRef");
+
+            if (!string.IsNullOrEmpty(rawVersionsBaseUrl) &&
+                !string.IsNullOrEmpty(buildInfoPath) &&
+                !string.IsNullOrEmpty(currentRef))
+            {
+                return BuildInfo.CachedGet(
+                    item.ItemSpec,
+                    rawVersionsBaseUrl,
+                    currentRef,
+                    buildInfoPath,
+                    cacheDir);
+            }
+
+            string packageListPath = item.GetMetadata("PackageListPath");
+
+            if (!string.IsNullOrEmpty(packageListPath))
+            {
+                using (var reader = new StreamReader(File.OpenRead(packageListPath)))
+                {
+                    return new BuildInfo
+                    {
+                        Name = item.ItemSpec,
+                        LatestPackages = BuildInfo.ReadPackageListAsync(reader).Result
+                    };
+                }
+            }
+
+            string packageId = item.GetMetadata("PackageId");
```

Comment:
This way lets you set the version via a msbuild property (like `microsoft.xunit.runner.uwp` in the example config PR). There's no benefit to the static file besides conciseness vs. direct msbuild, so that could be taken out.